lib/concurrent/synchronization/immutable_struct.rb

```diff
@@ -0,0 +1,67 @@
+module Concurrent
+  module Synchronization
+    # Similar to Struct but the fields are immutable and always visible (like Java final fields)
+    # @example
+    #   Person = ImmutableStruct.with_fields :name, :age
+    #   Person.new 'John Doe', 15
+    #   Person['John Doe', 15]
+    #   Person['John Doe', 15].members # => [:name, :age]
+    #   Person['John Doe', 15].values  # => ['John Doe', 15]
+    class ImmutableStruct < Synchronization::Object
+      def self.with_fields(*names, &block)
```

Comment:
I think construction and behavior of these should mirror Ruby's `Struct` class. Again, I'm a big fan of the concept but we should be aware of the user's expectations. This is similar to the conversation regarding `Agent`. `Struct` is part of the Ruby standard library so it's reasonable for a user to expect an `ImmutableStruct` to be "an immutable Struct."